src/System.Linq.Expressions/src/System/Runtime/CompilerServices/CallSite.cs

```diff
@@ -92,26 +92,21 @@ public static CallSite Create(Type delegateType, CallSiteBinder binder)
                 s_siteCtors = ctors = new CacheDict<Type, Func<CallSiteBinder, CallSite>>(100);
             }
 
-            Func<CallSiteBinder, CallSite> ctor;
-            MethodInfo method = null;
-            if (!ctors.TryGetValue(delegateType, out ctor))
+            if (!ctors.TryGetValue(delegateType, out Func<CallSiteBinder, CallSite> ctor))
             {
-                method = typeof(CallSite<>).MakeGenericType(delegateType).GetMethod(nameof(Create));
+                MethodInfo method = typeof(CallSite<>).MakeGenericType(delegateType).GetMethod(nameof(Create));
 
-                if (delegateType.CanCache())
+                if (delegateType.IsCollectible)
```

Comment:
This case in particular goes from 0% considered collectible since the delegates always have a `CallSite` type parameter that is guessed not collectible (though guessed collectible in netfx) to considering most common uses cacheable.